flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/InputBuffersMetricsTest.java

```diff
@@ -0,0 +1,301 @@
+	public void testCalculateTotalBuffersSize() throws IOException {
+		int numberOfRemoteChannels = 2;
+		int numberOfLocalChannels = 0;
+
+		int buffersPerChannel = 2;
+		int extraNetworkBuffersPerGate = 8;
+
+		CloseableRegistry closeableRegistry = new CloseableRegistry();
+
+		NettyShuffleEnvironment network = new NettyShuffleEnvironmentBuilder()
+			.setNetworkBuffersPerChannel(buffersPerChannel)
+			.setFloatingNetworkBuffersPerGate(extraNetworkBuffersPerGate)
+			.build();
+
+		Tuple3<SingleInputGate, List<RemoteInputChannel>, List<LocalInputChannel>> tuple1 = buildInputGate(
+			network,
+			numberOfRemoteChannels,
+			numberOfLocalChannels);
+
+		SingleInputGate inputGate1 = tuple1.f0;
+
+		closeableRegistry.registerCloseable(network::close);
+		closeableRegistry.registerCloseable(inputGate1::close);
+
+		try {
+			SingleInputGate[] inputGates = new SingleInputGate[]{inputGate1};
+			FloatingBuffersUsageGauge floatingBuffersUsageGauge = new FloatingBuffersUsageGauge(inputGates);
+			ExclusiveBuffersUsageGauge exclusiveBuffersUsageGauge = new ExclusiveBuffersUsageGauge(inputGates);
+			CreditBasedInputBuffersUsageGauge inputBufferPoolUsageGauge = new CreditBasedInputBuffersUsageGauge(
+				floatingBuffersUsageGauge,
+				exclusiveBuffersUsageGauge,
+				inputGates);
+
+			assertEquals(extraNetworkBuffersPerGate, floatingBuffersUsageGauge.calculateTotalBuffers(inputGate1));
+			assertEquals(numberOfRemoteChannels * buffersPerChannel, exclusiveBuffersUsageGauge.calculateTotalBuffers(inputGate1));
+			assertEquals(numberOfRemoteChannels * buffersPerChannel + extraNetworkBuffersPerGate, inputBufferPoolUsageGauge.calculateTotalBuffers(inputGate1));
+		} finally {
+			closeableRegistry.close();
+		}
+	}
+
+	@Test
+	public void testExclusiveBuffersUsage() throws IOException {
+		int numberOfRemoteChannelsGate1 = 2;
+		int numberOfLocalChannelsGate1 = 0;
+		int numberOfRemoteChannelsGate2 = 1;
+		int numberOfLocalChannelsGate2 = 1;
+
+		int numberOfRemoteChannelsTotal = numberOfRemoteChannelsGate1 + numberOfRemoteChannelsGate2;
+		int numberOfInputGates = 2;
+
+		int buffersPerChannel = 2;
+		int extraNetworkBuffersPerGate = 8;
+
+		CloseableRegistry closeableRegistry = new CloseableRegistry();
+
+		NettyShuffleEnvironment network = new NettyShuffleEnvironmentBuilder()
+			.setNetworkBuffersPerChannel(buffersPerChannel)
+			.setFloatingNetworkBuffersPerGate(extraNetworkBuffersPerGate)
+			.build();
+
+		Tuple3<SingleInputGate, List<RemoteInputChannel>, List<LocalInputChannel>> tuple1 = buildInputGate(
+			network,
+			numberOfRemoteChannelsGate1,
+			numberOfLocalChannelsGate1);
+		Tuple3<SingleInputGate, List<RemoteInputChannel>, List<LocalInputChannel>> tuple2 = buildInputGate(
+			network,
+			numberOfRemoteChannelsGate2,
+			numberOfLocalChannelsGate2);
+
+		SingleInputGate inputGate1 = tuple1.f0;
+		SingleInputGate inputGate2 = tuple2.f0;
+
+		try {
+			assertEquals(tuple1.f1.size(), numberOfRemoteChannelsGate1 + numberOfLocalChannelsGate1);
```

Comment:
this assert is not very necessary here, because it is not brought by this change.

Reply:
What do you think this issue?